app/database/db.go

```diff
@@ -37,24 +37,26 @@ func Open(source interface{}) (*DB, error) {
 }
 
 func (conn *DB) inTransaction(txFunc func(*DB) error) (err error) {
+	var transactionStarted = false
 	var txDB = conn.db.Begin()
-	if err != nil {
-		return err
+	if txDB.Error != nil {
+		return txDB.Error
 	}
+	transactionStarted = true
 	defer func() {
 		if p := recover(); p != nil {
 			// ensure rollback is executed even in case of panic
-			txDB.Rollback()
+			if transactionStarted {
```

Comment:
How can we arrive here with `! transactionStarted` as the `defer` is defined after the "begin/return error"?

Reply:
This function is a closure, so it sees all the variables from the scope that defines it.

Reply:
You're right, we cannot. I'll fix it right now